astropy/utils/tests/test_data.py

```diff
@@ -205,9 +202,6 @@ def no_TemporaryDirectory(*args, **kwargs):
             files = set(d.iterdir())
             monkeypatch.setattr(os, "mkdir", no_mkdir)
             monkeypatch.setattr(tempfile, "mkdtemp", no_mkdtemp)
-            monkeypatch.setattr(
-                astropy.utils.data, "_SafeTemporaryDirectory", no_TemporaryDirectory
-            )
```

Comment:
We don't need to replace this monkeypatch because `tempfile.TemporaryDirectory` calls `tempfile.mkdtemp()` and that is already monkeypatched.